### ISSUE TITLE
NA CCI: tcp: cannot bind to specific address

### ISSUE THREAD

**Issue by soumagne** [mercury-hpc/mercury#153]
CCI currently does not support passing a specific IP to bind to at initialization time. This makes it difficult when using the TCP transport as the user needs to provide a separate CCI config file with that information, such as:
```
[server_tcp]
transport = tcp
interface = eth0
port = 11050
priority = 10
mtu = 9000
```
and use the `CCI_CONFIG` environment variable to point to that config file.

**Comment by scottatchley** [comment]
Hi Jerome,

The way to do this is in Mercury:

/* e.g. argv[2] is “eth0” and argv[3] is “5555” */
int ret, i;
cci_device_t * const * devices = NULL;
char *interface = NULL;


ret = cci_get_devices(&devices);
if (ret) { /* handle error */ }

for (i = 0; ; i++) {
    cci_device_t *device = devices[i];
    if (!device)
        break;

    if (!strcmp(argv[2], device->name))
        break; /* found it */
}

if (!device) {
    /* no device found, need to abort */
}

ret = cci_create_endpoint_at(device, argv[3], ...);

Scott

**Comment by soumagne** [comment]
Oh ok I see, I will give it a try then, thanks!

**Comment by scottatchley** [comment]
I just noticed that you will need to remove the device declaration out of the loop so that you can use it in cci_create_endpoint().

**Comment by soumagne** [comment]
I included the fix and am just making some tweaks now but seems to be fine, thanks!

**Comment by soumagne** [comment]
I can't seem to find how to get the loopback interface though? cci_get_devices does not return it?

**Comment by soumagne** [comment]
Fixed in 128752a47a69dec6af2e3cdc381975f4aff34d7a (not using loopback interface though for now)
